andela/bp-esa-backend
# API Documentation Swagger

#### What does this PR do?
-  Introduce '**ESA**' API documentation using swagger
-  Aims to make product management a breeze- helps other internal teams know the details of  resources, methods, and their associated requests and responses, making maintenance and updates quicker.

#### Description of Task to be completed?
- Developers and other user alike will be able to integrate the ESA API as quickly as possible.

#### How should this be manually tested?
clone the repo
- `cd` into the project dir
- git-checkout to branch `ch-api-documentation-swagger`
- install application requirements with `npm install`
- use the information in `.env.sample` to provide the necessary environment variables i `.env`
- use `npm run start:dev` to start the application
- note that the default port is `8000`
- access sample swagger api documentation from the browser
#### Any background context you want to provide?
> Documentation is one of the most important factors in determining an API’s success, as strong, easy-to-understand documentation makes API implementation a breeze, while confusing, out-of-sync, incomplete or convoluted documentation makes for an unwelcome adventure—one that usually leads to frustrated developers

> Since APIs are designed to be consumed, it is important to make sure that the client, or consumer, is able to quickly implement an API and understand what is happening with it. 

> Unfortunately, many APIs make implementation extremely difficult, defeating their very purpose. As we build out our API we want to ensure that we not only provide informational API documentation to help developers integrate/ debug connections, but also return back relevant data whenever a user makes a call—especially a call that fails.

> While having a well formatted, coherent body response in your API documentation is extremely important (we want something that can easily be deserialized, iterated and understood), we also want to provide developers with quick references as to what happened with the call, including the use of status codes. 

> And in the case of a failure, we will want to provide descriptive error messages that tell the client not just what went wrong, but how to fix it.

#### What are the relevant pivotal tracker stories?(if applicable)
 #160582419
